app/controllers/users_controller.rb

```diff
@@ -146,6 +146,29 @@ def username
     end
   end
 
+  def toggle_second_factor
```

Comment:
I am not a fan of "toggle" actions that rely on state, I also feel user controller is getting quite big, perhaps a second_factor controller with an update method and then caller tells it if it wants it enable or disabled.